riff-cli/cmd/init.go

```diff
@@ -68,110 +69,38 @@ func Init() (*cobra.Command, *options.InitOptions) {
-	var initShellCmd = &cobra.Command{
-		Use:   "shell",
-		Short: "Initialize a shell script function",
-		Long:  utils.InitShellCmdLong(),
-
-		RunE: func(cmd *cobra.Command, args []string) error {
-			err := initializers.Shell().Initialize(*initOptions)
-			if err != nil {
-				return err
-			}
-			return nil
-		},
-	}
-	return initShellCmd, initOptions
-}
-
-func InitNode(initOptions *options.InitOptions) (*cobra.Command, *options.InitOptions) {
-	var initNodeCmd = &cobra.Command{
-		Use:   "node",
-		Short: "Initialize a node.js function",
-		Long:  utils.InitNodeCmdLong(),
-
-		RunE: func(cmd *cobra.Command, args []string) error {
-			err := initializers.Node().Initialize(*initOptions)
-			if err != nil {
-				return err
-			}
-			return nil
-		},
-		Aliases: []string{"js"},
+func InitFunctionInvokers(initOptions *options.InitOptions) ([]*cobra.Command, error) {
+	functionInvokers, err := initializer.LoadFunctionInvokers()
+	if err != nil {
+		return nil, err
 	}
-	return initNodeCmd, initOptions
-}
-
-func InitPython(initOptions *options.InitOptions) (*cobra.Command, *options.InitOptions) {
-	var initPythonCmd = &cobra.Command{
-		Use:   "python",
-		Short: "Initialize a Python function",
-		Long:  utils.InitPythonCmdLong(),
-
-		RunE: func(cmd *cobra.Command, args []string) error {
-			initOptions.Handler,_ = cmd.Flags().GetString("handler")
-			if initOptions.Handler == "" {
-				initOptions.Handler = initOptions.FunctionName
-			}
-			err := initializers.Python().Initialize(*initOptions)
-			if err != nil {
-				return err
-			}
-			return nil
-		},
-	}
-	initPythonCmd.Flags().String("handler", "", "the name of the function handler")
-	return initPythonCmd, initOptions
-}
-
-func InitGo(initOptions *options.InitOptions) (*cobra.Command, *options.InitOptions) {
-	var initGoCmd = &cobra.Command{
-		Use:   "go",
-		Short: "Initialize a go plugin function",
-		Long:  utils.InitGoCmdLong(),
-
-		RunE: func(cmd *cobra.Command, args []string) error {
-			initOptions.Handler,_ = cmd.Flags().GetString("handler")
-			if initOptions.Handler == "" {
-				initOptions.Handler = initOptions.FunctionName
-			}
-			err := initializers.Go().Initialize(*initOptions)
-			if err != nil {
-				return err
-			}
-			return nil
-		},
+	var initFunctionInvokerCmds []*cobra.Command
+	for _, functionInvoker := range functionInvokers {
+		functionInvokerName := functionInvoker.ObjectMeta.Name
+		var initFunctionInvokerCmd = &cobra.Command{
+			Use:   functionInvokerName,
+			Short: fmt.Sprintf("Initialize a %s function", functionInvokerName),
+			// Long:  "TODO",
+			RunE: func(cmd *cobra.Command, args []string) error {
+				initOptions.FunctionInvokerName = functionInvokerName
+				return initializer.Initialize(*initOptions)
+			},
+		}
+		if functionInvoker.Spec.Properties.HandlerRequired {
+			initFunctionInvokerCmd.PersistentFlags().StringVar(&initOptions.Handler, "handler", "", "the name of the function handler (required for some function invokers)")
```

Comment:
This is not a persistent flag, just a regular flag

Reply:
I believe there would be great value in having the last argument to this call be a property of our CRD (explains what exactly a "handler" is for each language)

Reply:
done